src/main/kotlin/org/koitharu/kotatsu/parsers/site/mangareader/en/RizzComic.kt

```diff
@@ -1,58 +1,152 @@
 package org.koitharu.kotatsu.parsers.site.mangareader.en
 
-import org.koitharu.kotatsu.parsers.ErrorMessages
+import okhttp3.FormBody
+import okhttp3.Request
 import org.koitharu.kotatsu.parsers.MangaLoaderContext
 import org.koitharu.kotatsu.parsers.MangaSourceParser
 import org.koitharu.kotatsu.parsers.model.*
 import org.koitharu.kotatsu.parsers.site.mangareader.MangaReaderParser
-import org.koitharu.kotatsu.parsers.util.domain
-import org.koitharu.kotatsu.parsers.util.oneOrThrowIfMany
-import org.koitharu.kotatsu.parsers.util.parseHtml
+import org.koitharu.kotatsu.parsers.util.*
+import org.koitharu.kotatsu.parsers.util.json.getFloatOrDefault
+import org.koitharu.kotatsu.parsers.util.json.getStringOrNull
+import org.koitharu.kotatsu.parsers.util.json.mapJSON
 import java.util.*
 
 @MangaSourceParser("RIZZCOMIC", "RizzComic", "en")
 internal class RizzComic(context: MangaLoaderContext) :
-	MangaReaderParser(context, MangaSource.RIZZCOMIC, "rizzfables.com", pageSize = 50, searchPageSize = 20) {
+	MangaReaderParser(context, MangaSource.RIZZCOMIC, "rizzfables.com", pageSize = 50, searchPageSize = 20){
 	override val datePattern = "dd MMM yyyy"
 	override val listUrl = "/series"
-
-	override val availableSortOrders: Set<SortOrder> = EnumSet.of(SortOrder.ALPHABETICAL)
-	override val availableStates: Set<MangaState> = emptySet()
-	override val isMultipleTagsSupported = false
+	override val availableSortOrders: Set<SortOrder> = EnumSet.of(SortOrder.ALPHABETICAL, SortOrder.UPDATED, SortOrder.NEWEST, SortOrder.POPULARITY, SortOrder.ALPHABETICAL_DESC)
+	override val availableStates: Set<MangaState> = EnumSet.of(MangaState.ONGOING, MangaState.FINISHED, MangaState.PAUSED)
+	override val isMultipleTagsSupported = true
 	override val isSearchSupported = true
 	override val isTagsExclusionSupported = false
 
-	// TODO Query created in json
+	private val filterUrl = "/Index/filter_series"
+	private val searchUrl = "/Index/live_search"
+	private val seriesCode = "r2311170"
```

Comment:
they change this every week or so, so it would be better to fetch from the site, using `SuspendLazy`